### PR TITLE
rpc: attempt to limit sctp congestion with back pressure

### PR DESCRIPTION
**Pull request by edaniels**

### CODE REVIEW

File: rpc/wrtc_client_stream.go [hunk 1]
```diff
@@ -169,19 +170,24 @@ var maxRequestMessagePacketDataSize int
 func init() {
 	md, err := proto.Marshal(&webrtcpb.Request{
 		Stream: &webrtcpb.Stream{
-			Id: 1,
+			Id: math.MaxUint64,
```
Comment (nicksanford) [inline rpc/wrtc_client_stream.go:173]: [q] why are we changing this id?

File: rpc/js/src/BaseChannel.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export class BaseChannel {
     private closed = false;
     private closedReason?: Error;
 
-    protected maxDataChannelSize = 16384;
+    protected maxDataChannelSize = 65535;
```
Comment (nicksanford) [inline rpc/js/src/BaseChannel.ts:15]: [q] why are we changing this if it is a no op?

File: rpc/wrtc_client_stream.go [hunk 3]
```diff
@@ -169,19 +170,24 @@ var maxRequestMessagePacketDataSize int
...
 			},
 		},
 	})
 	if err != nil {
 		panic(err)
 	}
-	// max msg size - packet size - msg type size - proto padding (?)
-	maxRequestMessagePacketDataSize = maxDataChannelSize - len(md) - 1
+	// maxRequestMessagePacketDataSize = maxDataChannelSize - max proto request wrapper size
+	maxRequestMessagePacketDataSize = maxDataChannelSize - len(md)
```
Comment (nicksanford) [inline rpc/wrtc_client_stream.go:190]: [q] Why are we removing the `-1`?

File: rpc/wrtc_server_stream.go [hunk 4]
```diff
@@ -131,19 +132,22 @@ var maxResponseMessagePacketDataSize int
...
 			},
 		},
 	})
 	if err != nil {
 		panic(err)
 	}
-	// max msg size - packet size - msg type size - proto padding (?)
-	maxResponseMessagePacketDataSize = maxDataChannelSize - len(md) - 1
+	// maxResponseMessagePacketDataSize = maxDataChannelSize - max proto response wrapper size
+	maxResponseMessagePacketDataSize = maxDataChannelSize - len(md)
```
Comment (nicksanford) [inline rpc/wrtc_server_stream.go:150]: Why are we removing the `-1`?

File: rpc/wrtc_server_stream.go [hunk 5]
```diff
@@ -131,19 +132,22 @@ var maxResponseMessagePacketDataSize int
 func init() {
 	md, err := proto.Marshal(&webrtcpb.Response{
 		Stream: &webrtcpb.Stream{
-			Id: 1,
+			Id: math.MaxUint64,
```
Comment (nicksanford) [inline rpc/wrtc_server_stream.go:135]: [q] why are we changing this id?

File: rpc/wrtc_server_stream.go [hunk 6]
```diff
@@ -131,19 +132,22 @@ var maxResponseMessagePacketDataSize int
...
 		},
 		Type: &webrtcpb.Response_Message{
 			Message: &webrtcpb.ResponseMessage{
-				PacketMessage: &webrtcpb.PacketMessage{Eom: true},
+				PacketMessage: &webrtcpb.PacketMessage{
```
Comment (nicksanford) [inline rpc/wrtc_server_stream.go:139]: Why do we need to change this struct?

File: rpc/wrtc_client_stream.go [hunk 7]
```diff
@@ -169,19 +170,24 @@ var maxRequestMessagePacketDataSize int
 func init() {
 	md, err := proto.Marshal(&webrtcpb.Request{
 		Stream: &webrtcpb.Stream{
-			Id: 1,
+			Id: math.MaxUint64,
 		},
 		Type: &webrtcpb.Request_Message{
 			Message: &webrtcpb.RequestMessage{
-				PacketMessage: &webrtcpb.PacketMessage{Eom: true},
+				HasMessage: true,
```
Comment (nicksanford) [inline rpc/wrtc_client_stream.go:177]: Why do we need to change this struct?

**Review by nicksanford — APPROVED** [review]
Approving as I trust you know what you are doing, questions added for my own edification & so I can explain it to others who may ask.

File: rpc/js/src/BaseChannel.ts (hunk 2 above)
Comment (edaniels) [reply rpc/js/src/BaseChannel.ts:15]: consistency!

File: rpc/wrtc_client_stream.go (hunk 3 above)
Comment (edaniels) [reply rpc/wrtc_client_stream.go:190]: didn't do anything

File: rpc/wrtc_server_stream.go (hunk 4 above)
Comment (edaniels) [reply rpc/wrtc_server_stream.go:150]: didn't do anything

**Pull request merged**